subsys/bluetooth/host/classic/l2cap_br.c

```diff
@@ -2028,10 +5649,60 @@ void bt_l2cap_br_recv(struct bt_conn *conn, struct net_buf *buf)
 	 */
 	check_fixed_channel(chan);
 
-	chan->ops->recv(chan, buf);
+#if L2CAP_BR_RET_FC_ENABLE
+	if (BR_CHAN(chan)->rx.mode != BT_L2CAP_BR_LINK_MODE_BASIC) {
+		/* Add back HDR for FCS check */
+		net_buf_push(buf, sizeof(*hdr));
+		bt_l2cap_br_ret_fc_recv(BR_CHAN(chan), buf);
+	} else {
+#endif /* L2CAP_BR_RET_FC_ENABLE */
+		chan->ops->recv(chan, buf);
+#if L2CAP_BR_RET_FC_ENABLE
+	}
+#endif /* L2CAP_BR_RET_FC_ENABLE */
 	net_buf_unref(buf);
 }
 
+int bt_l2cap_br_chan_recv_complete(struct bt_l2cap_chan *chan)
+{
+#if L2CAP_BR_RET_FC_ENABLE
+	struct bt_l2cap_br_chan *br_chan;
+	int err = 0;
+
+	br_chan = BR_CHAN(chan);
+
+	LOG_DBG("Receiving completed on %p", br_chan);
+
+	if ((br_chan->rx.mode == BT_L2CAP_BR_LINK_MODE_ERET) &&
+	    atomic_test_and_clear_bit(br_chan->flags, L2CAP_FLAG_LOCAL_BUSY)) {
+
+		if (!atomic_test_bit(br_chan->flags, L2CAP_FLAG_LOCAL_BUSY_CHANGED) &&
+		    !atomic_test_and_set_bit(br_chan->flags, L2CAP_FLAG_SEND_FRAME_P)) {
+			/* State: XMIT
+			 * Local Busy Clears
+			 * Action: send RR(P=1)
+			 */
+			br_chan->retry_count = 1;
+
+			l2cap_br_start_timer(br_chan, false, true);
```

Comment:
Unless the function takes just a single boolean parameter and it's clear from the name what that parameter does, having boolean parameters in functions is a terrible idea. Can you honestly say you'd know what the `false` and `true` do just by looking at this function call, if you didn't see the definition of the function?

Reply:
Well.
I will change the type of second parameter to 
```
enum l2cap_br_timer_type {
	BT_L2CAP_BR_TIMER_RET,
	BT_L2CAP_BR_TIMER_MONITOR,
};
```
Then the function can be called likes as `l2cap_br_start_timer(br_chan, BT_L2CAP_BR_TIMER_MONITOR, true);`

Reply:
What does the third parameter do?

Reply:
It is flag whether restart the timer.